.github/workflows/codeql.yml

```diff
@@ -0,0 +1,105 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL Advanced"
+
+on:
+  push:
+    branches: [ "4.x", "*", "2.4", "3.4", "5.x", "master", "next" ]
+  pull_request:
+    branches: [ "4.x", "*", "2.4", "3.4", "5.x", "master", "next" ]
+  schedule:
+    - cron: '44 13 * * 6'
```

Comment:
`2.4` and `3.4` branches are not supported any more. No need to scan them, we do not accept patches there.
`next` and `master` just follows 5.x and 4.x automatically and needed for compatibility (old urls). No need to scan them too.